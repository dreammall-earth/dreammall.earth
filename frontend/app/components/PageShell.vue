<template>
  <v-app>
    <slot />
  </v-app>
</template>
