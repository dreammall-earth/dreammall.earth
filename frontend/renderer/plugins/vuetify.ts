// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { ThemeDefinition, createVuetify } from 'vuetify'
import { aliases as mdiAliases } from 'vuetify/iconsets/mdi'
import { VNumberInput } from 'vuetify/labs/VNumberInput'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

import { aliases as customAliases } from '#assets/icons'
import dark from '#assets/scss/dark.module.scss'
import light from '#assets/scss/light.module.scss'

function makeThemeFromCssModule(theme: CSSModuleClasses, isDark: boolean): ThemeDefinition {
  return {
    dark: isDark,
    colors: {
      background: theme.backgroundColor,
      surface: theme.surfaceColor,
      primary: theme.primaryColor,
      secondary: theme.secondaryColor,
      success: theme.successColor,
      info: theme.infoColor,
      warning: theme.warningColor,
      error: theme.errorColor,
      font: theme.fontColor,
      'dropdown-background': theme.dropdownBackgroundColor,
      'cockpit-highlight': theme.cockpitHighlightColor,
      'dm-panel-submit-color': theme.dmPanelSubmitColor,
      'dm-panel-submit-background-color': theme.dmPanelSubmitBackgroundColor,
      'dm-panel-text-input-color': theme.dmPanelTextInputColor,
      'dm-panel-call-action-button-color': theme.dmPanelCallActionButtonColor,
      'dm-panel-call-action-button-background-color': theme.dmPanelCallActionButtonBackgroundColor,
      'dm-panel-call-action-button-indicator-color': theme.dmPanelCallActionButtonIndicatorColor,
      'dm-panel-call-action-button-indicator-background-color':
        theme.dmPanelCallActionButtonIndicatorBackgroundColor,
      'dm-panel-reminder-text-color': theme.dmPanelReminderTextColor,
      'dm-panel-reminder-link-color': theme.dmPanelReminderLinkColor,
      'dm-switch-inactive-color': theme.dmSwitchInactiveColor,
      'dm-switch-active-color': theme.dmSwitchActiveColor,
    },
    variables: {
      'border-color': theme.borderColor,
      'border-opacity': theme.borderOpacity,
      // Vuetify colors don't support opacity, so we use variables instead.
      // Beware that HEX values won't work.
      'icon-background': theme.iconBackground,
      'bottom-menu-background': theme.bottomMenuBackground,
      'sidebar-background': theme.sidebarBackground,
      'cockpit-input-background': theme.cockpitInputBackground,
      'cockpit-element-background': theme.cockpitElementBackground,
      'cockpit-chip-background': theme.cockpitChipBackground,
      'cockpit-chip-background-2': theme.cockpitChipBackground2,
      'panel-background': theme.panelBackground,
      'panel-element-background': theme.panelElementBackground,
      'dm-panel-text-input-background-color': theme.dmPanelTextInputBackgroundColor,
      'dm-panel-background-color': theme.dmPanelBackgroundColor,
      'dm-panel-border-color': theme.dmPanelBorderColor,
      'dm-panel-reminder-text-background-color': theme.dmPanelReminderTextBackgroundColor,
      'dm-panel-reminder-link-background-color': theme.dmPanelReminderLinkBackgroundColor,
    },
  }
}

const themes = {
  light: makeThemeFromCssModule(light, false),
  dark: makeThemeFromCssModule(dark, true),
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    components: {
      VNumberInput,
    },
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
    icons: {
      aliases: { ...mdiAliases, ...customAliases },
    },
    theme: {
      defaultTheme: 'light',
      themes,
    },
  })
