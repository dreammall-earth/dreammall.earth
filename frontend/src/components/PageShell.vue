<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
