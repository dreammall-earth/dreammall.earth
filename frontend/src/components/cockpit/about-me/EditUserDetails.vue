<template>
  <CockpitCard narrow>
    <template #header>
      <div class="header">
        <button class="back" @click="$emit('back')"><v-icon icon="$back"></v-icon></button>
        <h2>{{ $t('cockpit.about-me.edit.title') }}</h2>
      </div>
    </template>
    <template #default>
      <v-form class="add-detail" @submit.prevent="addDetail">
        <v-select
          v-model="newDetail.category"
          flat
          rounded
          :items="detailCategories"
          class="select-category"
        >
          <template #selection="{ item }">
            <v-icon :icon="detailCategoryToIcon(item.value)"></v-icon>
          </template>
          <template #item="{ item, props: listProps }">
            <v-list-item v-bind="listProps" title="">
              <v-icon :icon="detailCategoryToIcon(item.value)"></v-icon>
            </v-list-item>
          </template>
        </v-select>
        <v-text-field
          v-model="newDetail.text"
          name="text"
          clearable
          rounded
          flat
          density="compact"
          variant="solo"
          class="add-detail-text"
          maxlength="60"
        ></v-text-field>
        <v-btn type="submit" :disabled="newDetail.text?.length === 0" variant="flat" rounded>
          <v-icon icon="mdi mdi-plus"></v-icon>
        </v-btn>
      </v-form>
      <!-- </div> -->
      <Details :details="props.details" editable @remove-detail="removeDetail" />
    </template>
  </CockpitCard>
</template>

<script lang="ts" setup>
import { reactive } from 'vue'

import CockpitCard from '#components/cockpit/cockpitCard/CockpitCard.vue'

import { detailCategories } from './detailCategories'
import { detailCategoryToIcon } from './detailCategoryToIcon'
import Details from './UserDetails.vue'

import type { UserDetail, SocialMedia, AddUserDetailInput } from '#stores/userStore'

const props = defineProps<{
  username: string
  name: string
  userImage?: string
  initials?: string
  introduction?: string
  details: UserDetail[]
  social: SocialMedia[]
}>()

const emit = defineEmits<{
  (e: 'back'): void
  (e: 'add-detail', detail: AddUserDetailInput): void
  (e: 'remove-detail', id: number): void
}>()

const newDetail = reactive<AddUserDetailInput>({
  category: 'work',
  text: '',
})

const addDetail = () => {
  emit('add-detail', newDetail)
}

const removeDetail = (id: number) => {
  emit('remove-detail', id)
}
</script>

<style scoped>
.back {
  width: 40px;
}

.header {
  display: flex;
  gap: 20px;
}

.add-detail {
  display: flex;
  gap: 8px;
  align-items: center;
  justify-content: center;

  &:deep(.v-input__details) {
    display: none;
  }
}

.select-category {
  max-width: 80px;

  &:deep(.v-field__outline) {
    display: none;
  }
}

.add-detail-text {
  &:deep(.v-field) {
    background: grey; /* TODO set correct color */
  }
}
</style>
