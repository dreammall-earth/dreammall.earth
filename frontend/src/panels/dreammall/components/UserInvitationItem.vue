<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
