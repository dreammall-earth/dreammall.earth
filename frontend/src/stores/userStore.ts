import { useMutation, useQuery } from '@vue/apollo-composable'
import { acceptHMRUpdate, defineStore } from 'pinia'
import { computed, ref, watch } from 'vue'

import {
  addSocialMediaMutation,
  AddSocialMediaInput,
  AddSocialMediaMutationResult,
  SocialMediaType,
} from '#mutations/addSocialMediaMutation'
import {
  addUserDetailMutation,
  AddUserDetailInput,
  AddUserDetailMutationResult,
} from '#mutations/addUserDetailMutation'
import { removeSocialMediaMutation } from '#mutations/removeSocialMediaMutation'
import { removeUserDetailMutation } from '#mutations/removeUserDetailMutation'
import {
  updateUserMutation,
  UpdateUserInput,
  UpdateUserMutationResult,
} from '#mutations/updateUserMutation'
import { currentUserQuery } from '#queries/currentUserQuery'
import { searchUsersQuery } from '#queries/searchUsersQuery'
import { getInitials } from '#src/utils/getInitials'

export type { AddSocialMediaInput, SocialMediaType } from '#mutations/addSocialMediaMutation'
export type { AddUserDetailInput } from '#mutations/addUserDetailMutation'

export type UserInTable = {
  id: number
  role: 'VIEWER' | 'MODERATOR'
  name: string
  username: string
}

export type MyTable = {
  id: number
  name: string
  public: boolean
  users: UserInTable[]
}

export type UserDetailCategory = 'place' | 'work' | 'language' | 'education' | 'feeling'

export type UserDetail = {
  id: number
  category: UserDetailCategory
  text: string
}

export type UserAvailability = null | 'available' | 'partly_available' | 'busy'

export type SocialMedia = {
  id: number
  type: SocialMediaType
  link: string
}

export type UserWithProfile = {
  id: number
  name: string
  username: string
  avatar?: string
  table?: MyTable
  introduction?: string
  availability: UserAvailability
  details: UserDetail[]
  social: SocialMedia[]
}

export type CurrentUser = UserWithProfile & { referenceId: string }

export type SearchUser = {
  id: number
  name: string
  username: string
}

export type SearchUsersResult = {
  users: SearchUser[]
}

export const useUserStore = defineStore(
  'user',
  () => {
    const currentUser = ref<CurrentUser | null>(null)

    const { result: currentUserQueryResult, loading } = useQuery(
      currentUserQuery,
      {},
      {
        prefetch: false,
        fetchPolicy: 'no-cache',
      },
    )

    watch(currentUserQueryResult, (data: { currentUser: CurrentUser }) => {
      setCurrentUser(data.currentUser)
    })

    const getCurrentUser = computed(() => currentUser.value)

    const getMyTable = computed(() => currentUser.value?.table)

    const setMyTable = (table: MyTable) => {
      if (currentUser.value) {
        setCurrentUser({
          ...currentUser.value,
          table,
        })
      }
    }

    const getUsersInMyTable = computed(() => currentUser.value?.table?.users)

    const getCurrentUserAvatar = computed(() => currentUser.value?.avatar)

    const getCurrentUserInitials = computed(() => {
      const name = currentUser.value?.name
      if (name) return getInitials(name)
      return ''
    })

    const setCurrentUser = (user: CurrentUser) => {
      currentUser.value = user
    }

    const { mutate: updateUserMutationResult } = useMutation<UpdateUserMutationResult>(
      updateUserMutation,
      {
        fetchPolicy: 'no-cache',
      },
    )

    const updateUser = async (data: UpdateUserInput) => {
      const result = await updateUserMutationResult({ data })
      if (result?.data) {
        setCurrentUser(result.data.updateUser)
      }
    }

    const { mutate: addUserDetailMutationResult } = useMutation<AddUserDetailMutationResult>(
      addUserDetailMutation,
      {
        fetchPolicy: 'no-cache',
      },
    )

    const addUserDetail = async (userDetail: AddUserDetailInput) => {
      const result = await addUserDetailMutationResult({ data: userDetail })
      if (result?.data && currentUser.value?.id) {
        setCurrentUser({
          ...currentUser.value,
          details: [...currentUser.value.details, result.data.addUserDetail],
        })
      }
    }

    const { mutate: removeUserDetailMutationResult } = useMutation(removeUserDetailMutation, {
      fetchPolicy: 'no-cache',
    })

    const removeUserDetail = async (userDetailId: number) => {
      await removeUserDetailMutationResult({ id: userDetailId })
      if (currentUser.value?.id) {
        setCurrentUser({
          ...currentUser.value,
          details: currentUser.value.details.filter((d) => d.id !== userDetailId),
        })
      }
    }

    const { mutate: AddSocialMediaMutationResult } = useMutation<AddSocialMediaMutationResult>(
      addSocialMediaMutation,
      {
        fetchPolicy: 'no-cache',
      },
    )

    const addSocialMedia = async (socialMedia: AddSocialMediaInput) => {
      const result = await AddSocialMediaMutationResult({ data: socialMedia })
      if (result?.data && currentUser.value?.id) {
        setCurrentUser({
          ...currentUser.value,
          social: [...currentUser.value.social, result.data.addSocialMedia],
        })
      }
    }

    const { mutate: removeSocialMediaMutationResult } = useMutation(removeSocialMediaMutation, {
      fetchPolicy: 'no-cache',
    })

    const removeSocialMedia = async (socialMediaId: number) => {
      await removeSocialMediaMutationResult({ id: socialMediaId })
      if (currentUser.value?.id) {
        setCurrentUser({
          ...currentUser.value,
          social: currentUser.value.social.filter((s) => s.id !== socialMediaId),
        })
      }
    }

    const searchUsers = (name: string) => {
      const { result, loading, error } = useQuery(
        searchUsersQuery,
        { searchString: name },
        {
          fetchPolicy: 'no-cache',
        },
      )
      return { result, loading, error }
    }

    return {
      loading,
      currentUser,
      getCurrentUser,
      setCurrentUser,
      getCurrentUserInitials,
      getCurrentUserAvatar,
      getMyTable,
      setMyTable,
      getUsersInMyTable,
      searchUsers,
      updateUser,
      addUserDetail,
      addSocialMedia,
      removeUserDetail,
      removeSocialMedia,
    }
  },
  {
    persist: false,
  },
)

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useUserStore, import.meta.hot))
}
