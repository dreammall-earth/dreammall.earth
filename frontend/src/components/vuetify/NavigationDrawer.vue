<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
