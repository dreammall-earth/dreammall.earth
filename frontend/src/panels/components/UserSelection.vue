<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
