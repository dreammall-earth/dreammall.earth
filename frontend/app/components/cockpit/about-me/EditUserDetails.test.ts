import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import { UserDetail } from '#app/stores/userStore'

import EditUserDetails from './EditUserDetails.vue'

describe('EditUserDetails', () => {
  const Wrapper = (
    details = [
      {
        id: 1,
        category: 'education',
        text: 'I am a student',
      },
      {
        id: 2,
        category: 'work',
        text: 'I am a worker',
      },
      {
        id: 3,
        category: 'work',
        text: 'I really like my job',
      },
      {
        id: 4,
        category: 'language',
        text: 'English',
      },
      {
        id: 5,
        category: 'place',
        text: 'Berlin',
      },
      {
        id: 6,
        category: 'feeling',
        text: 'Yeah!',
      },
    ] as UserDetail[],
  ) =>
    mount(VApp, {
      slots: {
        default: h(EditUserDetails as Component, {
          details,
        }),
      },
    })

  it('renders', () => {
    const wrapper = Wrapper()
    expect(wrapper.element).toMatchSnapshot()
  })
})
