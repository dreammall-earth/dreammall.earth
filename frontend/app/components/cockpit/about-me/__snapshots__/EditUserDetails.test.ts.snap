// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditUserDetails > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    <section
      class="cockpit-card"
      data-v-7779afd2=""
      data-v-eb62fc14=""
    >
      <div
        class="header"
        data-v-7779afd2=""
      >
        
        <div
          class="header"
          data-v-eb62fc14=""
        >
          <button
            class="back"
            data-v-eb62fc14=""
          >
            <i
              aria-hidden="true"
              class="mdi mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default text-icon"
              data-v-eb62fc14=""
            />
          </button>
          <h2
            data-v-eb62fc14=""
          >
            $t('cockpit.about-me.edit.title')
          </h2>
        </div>
        
      </div>
      <hr
        class="hr"
        data-v-7779afd2=""
      />
      
      <form
        class="v-form add-detail mb-2"
        data-v-eb62fc14=""
        novalidate=""
      >
        
        <div
          class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected v-select--selection-slot select-category"
          data-v-eb62fc14=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--flat v-field--no-label v-field--variant-filled v-theme--light v-field--rounded v-locale--is-ltr"
              role="combobox"
            >
              <div
                class="v-field__overlay"
              />
              <div
                class="v-field__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="opacity: NaN;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="opacity: NaN; width: 0%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div
                class="v-field__field"
                data-no-activator=""
              >
                <!---->
                <label
                  class="v-label v-field-label"
                  for="input-0"
                >
                  <!---->
                  
                  <!---->
                  
                </label>
                
                
                <!---->
                <div
                  class="v-field__input"
                  data-no-activator=""
                >
                  
                  
                  
                  <!---->
                  <!---->
                  
                  
                  <div
                    class="v-select__selection"
                  >
                    
                    <i
                      aria-hidden="true"
                      class="mdi mdi-briefcase-outline mdi v-icon notranslate v-theme--light v-icon--size-default"
                      data-v-eb62fc14=""
                    />
                    
                  </div>
                  
                  
                  
                  <input
                    aria-describedby="input-0-messages"
                    aria-label="Öffnen"
                    id="input-0"
                    inputmode="none"
                    size="1"
                    title="Öffnen"
                    type="text"
                    value="work"
                  />
                </div>
                <!---->
                
                
              </div>
              <!---->
              <div
                class="v-field__append-inner"
              >
                
                
                <!---->
                <i
                  aria-hidden="true"
                  class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon"
                />
                
                
                <!---->
              </div>
              <div
                class="v-field__outline"
              >
                <!---->
                <!---->
              </div>
            </div>
            
          </div>
          <!---->
          <div
            class="v-input__details"
          >
            <transition-group-stub
              appear="false"
              aria-live="polite"
              class="v-messages"
              css="true"
              id="input-0-messages"
              name="slide-y-transition"
              persisted="false"
              role="alert"
              tag="div"
            >
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div
          class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-text-field add-detail-text"
          data-v-eb62fc14=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-field v-field--appended v-field--center-affix v-field--flat v-field--no-label v-field--variant-solo v-theme--light v-field--rounded v-locale--is-ltr"
            >
              <div
                class="v-field__overlay"
              />
              <div
                class="v-field__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="opacity: NaN;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="opacity: NaN; width: 0%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div
                class="v-field__field"
                data-no-activator=""
              >
                <!---->
                <label
                  class="v-label v-field-label"
                  for="input-3"
                >
                  <!---->
                  
                  <!---->
                  
                </label>
                
                
                <!---->
                <input
                  aria-describedby="input-3-messages"
                  class="v-field__input"
                  id="input-3"
                  maxlength="60"
                  name="text"
                  size="1"
                  type="text"
                  value=""
                />
                <!---->
                
                
              </div>
              <transition-stub
                appear="false"
                css="true"
                name="expand-x-transition"
                persisted="false"
              >
                <div
                  class="v-field__clearable"
                  style="display: none;"
                >
                  
                  <i
                    aria-hidden="false"
                    aria-label=" leeren"
                    class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable"
                    role="button"
                    tabindex="0"
                  />
                  
                </div>
              </transition-stub>
              <div
                class="v-field__append-inner"
              >
                <!---->
                <!---->
              </div>
              <div
                class="v-field__outline"
              >
                <!---->
                <!---->
              </div>
            </div>
            
          </div>
          <!---->
          <div
            class="v-input__details"
          >
            <transition-group-stub
              appear="false"
              aria-live="polite"
              class="v-messages"
              css="true"
              id="input-3-messages"
              name="slide-y-transition"
              persisted="false"
              role="alert"
              tag="div"
            >
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <button
          class="submit rounded-circle"
          data-v-eb62fc14=""
          disabled=""
          type="submit"
        >
          <i
            aria-hidden="true"
            class="mdi mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default"
            data-v-eb62fc14=""
          />
        </button>
        
      </form>
      <!-- &lt;/div&gt; -->
      <ul
        class="details editable"
        data-v-eb62fc14=""
        data-v-f9dbb6f1=""
      >
        
        <li
          data-v-f9dbb6f1=""
        >
          <span
            class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal detail"
            data-v-f9dbb6f1=""
            draggable="false"
          >
            
            <!---->
            <span
              class="v-chip__underlay"
            />
            
            <!---->
            <!---->
            <div
              class="v-chip__content"
              data-no-activator=""
            >
              
              <i
                aria-hidden="true"
                class="mdi mdi-school-outline mdi v-icon notranslate v-theme--light v-icon--size-default text-cockpit-highlight mr-2"
                data-v-f9dbb6f1=""
              />
               I am a student 
              <button
                data-v-f9dbb6f1=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                  data-v-f9dbb6f1=""
                />
              </button>
              
            </div>
            <!---->
            <!---->
          </span>
        </li>
        <li
          data-v-f9dbb6f1=""
        >
          <span
            class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal detail"
            data-v-f9dbb6f1=""
            draggable="false"
          >
            
            <!---->
            <span
              class="v-chip__underlay"
            />
            
            <!---->
            <!---->
            <div
              class="v-chip__content"
              data-no-activator=""
            >
              
              <i
                aria-hidden="true"
                class="mdi mdi-briefcase-outline mdi v-icon notranslate v-theme--light v-icon--size-default text-cockpit-highlight mr-2"
                data-v-f9dbb6f1=""
              />
               I am a worker 
              <button
                data-v-f9dbb6f1=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                  data-v-f9dbb6f1=""
                />
              </button>
              
            </div>
            <!---->
            <!---->
          </span>
        </li>
        <li
          data-v-f9dbb6f1=""
        >
          <span
            class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal detail"
            data-v-f9dbb6f1=""
            draggable="false"
          >
            
            <!---->
            <span
              class="v-chip__underlay"
            />
            
            <!---->
            <!---->
            <div
              class="v-chip__content"
              data-no-activator=""
            >
              
              <i
                aria-hidden="true"
                class="mdi mdi-briefcase-outline mdi v-icon notranslate v-theme--light v-icon--size-default text-cockpit-highlight mr-2"
                data-v-f9dbb6f1=""
              />
               I really like my job 
              <button
                data-v-f9dbb6f1=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                  data-v-f9dbb6f1=""
                />
              </button>
              
            </div>
            <!---->
            <!---->
          </span>
        </li>
        <li
          data-v-f9dbb6f1=""
        >
          <span
            class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal detail"
            data-v-f9dbb6f1=""
            draggable="false"
          >
            
            <!---->
            <span
              class="v-chip__underlay"
            />
            
            <!---->
            <!---->
            <div
              class="v-chip__content"
              data-no-activator=""
            >
              
              <i
                aria-hidden="true"
                class="mdi mdi-web mdi v-icon notranslate v-theme--light v-icon--size-default text-cockpit-highlight mr-2"
                data-v-f9dbb6f1=""
              />
               English 
              <button
                data-v-f9dbb6f1=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                  data-v-f9dbb6f1=""
                />
              </button>
              
            </div>
            <!---->
            <!---->
          </span>
        </li>
        <li
          data-v-f9dbb6f1=""
        >
          <span
            class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal detail"
            data-v-f9dbb6f1=""
            draggable="false"
          >
            
            <!---->
            <span
              class="v-chip__underlay"
            />
            
            <!---->
            <!---->
            <div
              class="v-chip__content"
              data-no-activator=""
            >
              
              <i
                aria-hidden="true"
                class="mdi mdi-map-marker-outline mdi v-icon notranslate v-theme--light v-icon--size-default text-cockpit-highlight mr-2"
                data-v-f9dbb6f1=""
              />
               Berlin 
              <button
                data-v-f9dbb6f1=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                  data-v-f9dbb6f1=""
                />
              </button>
              
            </div>
            <!---->
            <!---->
          </span>
        </li>
        <li
          data-v-f9dbb6f1=""
        >
          <span
            class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal detail"
            data-v-f9dbb6f1=""
            draggable="false"
          >
            
            <!---->
            <span
              class="v-chip__underlay"
            />
            
            <!---->
            <!---->
            <div
              class="v-chip__content"
              data-no-activator=""
            >
              
              <i
                aria-hidden="true"
                class="mdi mdi-emoticon-outline mdi v-icon notranslate v-theme--light v-icon--size-default text-cockpit-highlight mr-2"
                data-v-f9dbb6f1=""
              />
               Yeah! 
              <button
                data-v-f9dbb6f1=""
              >
                <i
                  aria-hidden="true"
                  class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default"
                  data-v-f9dbb6f1=""
                />
              </button>
              
            </div>
            <!---->
            <!---->
          </span>
        </li>
        
      </ul>
      
    </section>
    
  </div>
</div>
`;
